packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -120,10 +122,18 @@ export class DataStores implements IDisposable {
                 }
 
                 /**
-                 * If there is no isRootDataStore in the attributes blob, set it to true. This will ensure that data
-                 * stores in older documents are not garbage collected incorrectly. This may lead to additional roots
-                 * in the document but they won't break.
+                 * Read the GC blob from summary. If there is no GC blob, set isRootDataStore to true. This will
+                 * ensure that data stores in older documents are not garbage collected incorrectly. This may lead
+                 * to additional roots in the document but they won't break.
                  */
+                let isRootDataStore = true;
+                if (snapshotTree.blobs[gcBlobKey] !== undefined) {
```

Comment:
If we always read this blob on boot, then I believe there is no reason for this data not to be rolled into .attributes.
We either need to do that, or separate isRoot to be in .attributes, and gc blob be only about info that only summarizer needs (i.e. can be loaded on demand)

Reply:
This is only for detached data stores. Remote data stores don't need to read this blob on boot. They can generate this on demand, which will only be during GC / summary.

Reply:
Any way to ensure that isRoot() is not called on load path? With GC data I have less concern, as it's more obvious, but asking if it's root - I feel like we will build feature that required this on boot (for example, getRootDataStore is used today on boot, but really does not check if data store is root, as we never had that data. With this addition, it's natural to add it (once we solve some other unrelated problems in Bohemia :)).
So maybe root boolean should be part of attributes, as it is always known and part of the data store core state, but GC is something optional (to an extent).

Reply:
You are right. isRoot() can be called in other places as well. And it makes sense for it to be part of attributes because it seems like a part of the data store itself rather than being GC specific.